MigrationInvocationsSafetyTest.migrationCommit_shouldRollback_whenTargetCrashes failure

*master*

Failed on Oracle JDK 1.6: https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-OracleJDK1.6-freestyle/427/

Stacktrace:
```
java.lang.AssertionError: expected:<PartitionTable{addresses=[[[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null], [[127.0.0.1]:20849, null, null, null, null, null, null]], version=632}> but was:<PartitionTable{addresses=[[[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null], [[127.0.0.1]:20850, null, null, null, null, null, null]], version=631}>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at com.hazelcast.internal.partition.MigrationInvocationsSafetyTest$8.run(MigrationInvocationsSafetyTest.java:286)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1242)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1259)
	at com.hazelcast.internal.partition.MigrationInvocationsSafetyTest.migrationCommit_shouldRollback_whenTargetCrashes(MigrationInvocationsSafetyTest.java:283)
```

Standard output:
```
[dev] [3.10-SNAPSHOT] MemberInfo{address=[127.0.0.1]:20852, uuid=862b1c14-e567-4582-964c-224b3c76e9de, liteMember=false, memberListJoinVersion=4} is excluded because I couldn't get its acceptance
java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: [127.0.0.1]:20851 claims mastership but it is not a member!
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:92) ~[classes/:?]
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:154) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.MembershipManager.addAcceptedMemberInfo(MembershipManager.java:783) [classes/:?]
	at com.hazelcast.internal.cluster.impl.MembershipManager.decideNewMembersView(MembershipManager.java:774) [classes/:?]
	at com.hazelcast.internal.cluster.impl.MembershipManager.access$100(MembershipManager.java:80) [classes/:?]
	at com.hazelcast.internal.cluster.impl.MembershipManager$DecideNewMembersViewTask.run(MembershipManager.java:1199) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:439) [?:1.6.0_45]
	at com.hazelcast.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:67) [classes/:?]
	at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:227) [classes/:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:895) [?:1.6.0_45]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:918) [?:1.6.0_45]
	at java.lang.Thread.run(Thread.java:662) [?:1.6.0_45]
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) [classes/:?]
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) [classes/:?]
Caused by: java.lang.IllegalArgumentException: [127.0.0.1]:20851 claims mastership but it is not a member!
	at com.hazelcast.util.Preconditions.checkTrue(Preconditions.java:327) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.ClusterServiceImpl.handleMastershipClaim(ClusterServiceImpl.java:262) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.operations.FetchMembersViewOp.run(FetchMembersViewOp.java:58) ~[classes/:?]
	at com.hazelcast.spi.Operation.call(Operation.java:148) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:202) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:191) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:405) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:115) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:100) ~[classes/:?]
	at ------ submitted from ------.(Unknown Source) ~[?:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:127) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:79) ~[classes/:?]
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:190) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.MembershipManager.fetchLatestMembersView(MembershipManager.java:818) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.MembershipManager.decideNewMembersView(MembershipManager.java:742) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.MembershipManager.access$100(MembershipManager.java:80) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.MembershipManager$DecideNewMembersViewTask.run(MembershipManager.java:1199) ~[classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:439) ~[?:1.6.0_45]
	at com.hazelcast.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:67) ~[classes/:?]
	at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:227) ~[classes/:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:895) ~[?:1.6.0_45]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:918) ~[?:1.6.0_45]
	at java.lang.Thread.run(Thread.java:662) ~[?:1.6.0_45]
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) ~[classes/:?]
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) ~[classes/:?]
	at ------ submitted from ------.(Unknown Source) ~[?:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:127) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:79) ~[classes/:?]
	... 13 more
```

*maintenance* 

New failure on PR builder: https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/new-lab-fast-pr/15425/testReport/junit/com.hazelcast.internal.partition/MigrationInvocationsSafetyTest/migrationCommit_shouldBeRetried_whenTargetNotResponds/

Stacktrace:
```
java.lang.AssertionError: Instances not in safe state! {[127.0.0.1]:5001=REPLICA_NOT_OWNED, [127.0.0.1]:5003=REPLICA_NOT_OWNED, [127.0.0.1]:5002=REPLICA_NOT_OWNED, [127.0.0.1]:5004=REPLICA_NOT_OWNED}
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.test.HazelcastTestSupport.assertAllInSafeState(HazelcastTestSupport.java:768)
	at com.hazelcast.test.HazelcastTestSupport$4.run(HazelcastTestSupport.java:733)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1244)
	at com.hazelcast.test.HazelcastTestSupport.waitAllForSafeState(HazelcastTestSupport.java:731)
	at com.hazelcast.test.HazelcastTestSupport.waitAllForSafeState(HazelcastTestSupport.java:727)
	at com.hazelcast.test.HazelcastTestSupport.waitAllForSafeState(HazelcastTestSupport.java:753)
	at com.hazelcast.internal.partition.MigrationInvocationsSafetyTest.migrationCommit_shouldBeRetried_whenTargetNotResponds(MigrationInvocationsSafetyTest.java:231)
```

Failed again on a Windows maintenance build
```java
java.lang.AssertionError: expected:<1> but was:<2>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.internal.partition.MigrationInvocationsSafetyTest.assertNoDuplicateMigrations(MigrationInvocationsSafetyTest.java:449)
	at com.hazelcast.internal.partition.MigrationInvocationsSafetyTest.migrationCommit_shouldRollback_whenTargetCrashes(MigrationInvocationsSafetyTest.java:330)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:105)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:97)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.maintenance-Windows-OracleJDK8/5/testReport/junit/com.hazelcast.internal.partition/MigrationInvocationsSafetyTest/migrationCommit_shouldRollback_whenTargetCrashes/

Failed again here
http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/4764/testReport/junit/com.hazelcast.internal.partition/MigrationInvocationsSafetyTest/promotionCommit_shouldRollback_whenTargetCrashes/

```
Error Message
Partition: 14, replica: 3 data is missing on [127.0.0.1]:5703 - d9fe4066-6ec3-4f4f-a588-be470d0e383a
Stacktrace
java.lang.AssertionError: Partition: 14, replica: 3 data is missing on [127.0.0.1]:5703 - d9fe4066-6ec3-4f4f-a588-be470d0e383a
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.internal.partition.PartitionCorrectnessTestSupport.assertNoMissingData(PartitionCorrectnessTestSupport.java:280)
	at com.hazelcast.internal.partition.PartitionCorrectnessTestSupport.assertSizeAndData(PartitionCorrectnessTestSupport.java:244)
	at com.hazelcast.internal.partition.PartitionCorrectnessTestSupport.access$000(PartitionCorrectnessTestSupport.java:63)
	at com.hazelcast.internal.partition.PartitionCorrectnessTestSupport$AssertSizeAndDataTask.run(PartitionCorrectnessTestSupport.java:389)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1326)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1428)
	at com.hazelcast.internal.partition.PartitionCorrectnessTestSupport.assertSizeAndDataEventually(PartitionCorrectnessTestSupport.java:206)
	at com.hazelcast.internal.partition.PartitionCorrectnessTestSupport.assertSizeAndDataEventually(PartitionCorrectnessTestSupport.java:202)
	at com.hazelcast.internal.partition.MigrationInvocationsSafetyTest.promotionCommit_shouldRollback_whenTargetCrashes(MigrationInvocationsSafetyTest.java:411)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/hazelcast/src/test/java/com/hazelcast/internal/partition/MigrationInvocationsSafetyTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/partition/MigrationInvocationsSafetyTest.java
@@ -19,17 +19,15 @@ package com.hazelcast.internal.partition;
 import com.hazelcast.cluster.ClusterState;
 import com.hazelcast.config.Config;
 import com.hazelcast.core.HazelcastInstance;
-import com.hazelcast.internal.partition.impl.MigrationInterceptor;
 import com.hazelcast.internal.partition.impl.InternalPartitionServiceImpl;
+import com.hazelcast.internal.partition.impl.MigrationInterceptor;
 import com.hazelcast.internal.partition.service.TestMigrationAwareService;
 import com.hazelcast.spi.impl.SpiDataSerializerHook;
 import com.hazelcast.spi.properties.ClusterProperty;
-import com.hazelcast.test.AssertTask;
 import com.hazelcast.test.ChangeLoggingRule;
 import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.annotation.ParallelJVMTest;
 import com.hazelcast.test.annotation.QuickTest;
-import org.hamcrest.Matchers;
 import org.junit.Before;
 import org.junit.ClassRule;
 import org.junit.Test;
@@ -76,11 +74,11 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
 
     @Test
     public void members_shouldAgree_onPartitionTable_whenMasterChanges() {
-        final HazelcastInstance initialMaster = factory.newHazelcastInstance();
-        final HazelcastInstance nextMaster = factory.newHazelcastInstance();
-        final HazelcastInstance slave1 = factory.newHazelcastInstance();
-        final HazelcastInstance slave2 = factory.newHazelcastInstance();
-        final HazelcastInstance slave3 = factory.newHazelcastInstance();
+        HazelcastInstance initialMaster = factory.newHazelcastInstance();
+        HazelcastInstance nextMaster = factory.newHazelcastInstance();
+        HazelcastInstance slave1 = factory.newHazelcastInstance();
+        HazelcastInstance slave2 = factory.newHazelcastInstance();
+        HazelcastInstance slave3 = factory.newHazelcastInstance();
 
         assertClusterSizeEventually(5, nextMaster, slave1, slave2, slave3);
 
@@ -101,36 +99,30 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
 
         terminateInstance(initialMaster);
 
-        spawn(new Runnable() {
-            @Override
-            public void run() {
-                assertClusterSizeEventually(4, nextMaster, slave1, slave2, slave3);
-                sleepSeconds(10);
-                resetPacketFiltersFrom(nextMaster);
-            }
+        spawn(() -> {
+            assertClusterSizeEventually(4, nextMaster, slave1, slave2, slave3);
+            sleepSeconds(10);
+            resetPacketFiltersFrom(nextMaster);
         });
 
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() {
-                int nextPartitionStateVersion = getPartitionService(nextMaster).getPartitionStateVersion();
-                assertThat(nextPartitionStateVersion, greaterThan(initialPartitionStateVersion));
+        assertTrueEventually(() -> {
+            int nextPartitionStateVersion = getPartitionService(nextMaster).getPartitionStateVersion();
+            assertThat(nextPartitionStateVersion, greaterThan(initialPartitionStateVersion));
 
-                assertEquals(nextPartitionStateVersion, getPartitionService(slave1).getPartitionStateVersion());
-                assertEquals(nextPartitionStateVersion, getPartitionService(slave2).getPartitionStateVersion());
-                assertEquals(nextPartitionStateVersion, getPartitionService(slave3).getPartitionStateVersion());
-            }
+            assertEquals(nextPartitionStateVersion, getPartitionService(slave1).getPartitionStateVersion());
+            assertEquals(nextPartitionStateVersion, getPartitionService(slave2).getPartitionStateVersion());
+            assertEquals(nextPartitionStateVersion, getPartitionService(slave3).getPartitionStateVersion());
         });
 
     }
 
     @Test
     public void members_shouldAgree_onPartitionTable_whenMasterChanges_and_anotherMemberCrashes() {
-        final HazelcastInstance initialMaster = factory.newHazelcastInstance();
-        final HazelcastInstance nextMaster = factory.newHazelcastInstance();
-        final HazelcastInstance slave1 = factory.newHazelcastInstance();
-        final HazelcastInstance slave2 = factory.newHazelcastInstance();
-        final HazelcastInstance slave3 = factory.newHazelcastInstance();
+        HazelcastInstance initialMaster = factory.newHazelcastInstance();
+        HazelcastInstance nextMaster = factory.newHazelcastInstance();
+        HazelcastInstance slave1 = factory.newHazelcastInstance();
+        HazelcastInstance slave2 = factory.newHazelcastInstance();
+        HazelcastInstance slave3 = factory.newHazelcastInstance();
 
         assertClusterSizeEventually(5, nextMaster, slave1, slave2, slave3);
 
@@ -151,24 +143,18 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
 
         terminateInstance(initialMaster);
 
-        spawn(new Runnable() {
-            @Override
-            public void run() {
-                assertClusterSizeEventually(4, nextMaster, slave1, slave2, slave3);
-                sleepSeconds(10);
-                terminateInstance(slave3);
-            }
+        spawn(() -> {
+            assertClusterSizeEventually(4, nextMaster, slave1, slave2, slave3);
+            sleepSeconds(10);
+            terminateInstance(slave3);
         });
 
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() {
-                int nextPartitionStateVersion = getPartitionService(nextMaster).getPartitionStateVersion();
-                assertThat(nextPartitionStateVersion, greaterThan(initialPartitionStateVersion));
+        assertTrueEventually(() -> {
+            int nextPartitionStateVersion = getPartitionService(nextMaster).getPartitionStateVersion();
+            assertThat(nextPartitionStateVersion, greaterThan(initialPartitionStateVersion));
 
-                assertEquals(nextPartitionStateVersion, getPartitionService(slave1).getPartitionStateVersion());
-                assertEquals(nextPartitionStateVersion, getPartitionService(slave2).getPartitionStateVersion());
-            }
+            assertEquals(nextPartitionStateVersion, getPartitionService(slave1).getPartitionStateVersion());
+            assertEquals(nextPartitionStateVersion, getPartitionService(slave2).getPartitionStateVersion());
         });
 
     }
@@ -184,9 +170,9 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
     }
 
     private void partitionState_shouldNotBeSafe_duringPartitionTableFetch_whenMasterChanges(boolean liteMaster) {
-        final HazelcastInstance initialMaster = factory.newHazelcastInstance(new Config().setLiteMember(liteMaster));
-        final HazelcastInstance nextMaster = factory.newHazelcastInstance();
-        final HazelcastInstance slave = factory.newHazelcastInstance();
+        HazelcastInstance initialMaster = factory.newHazelcastInstance(new Config().setLiteMember(liteMaster));
+        HazelcastInstance nextMaster = factory.newHazelcastInstance();
+        HazelcastInstance slave = factory.newHazelcastInstance();
 
         assertClusterSizeEventually(3, nextMaster, slave);
         warmUpPartitions(initialMaster, nextMaster, slave);
@@ -196,22 +182,16 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
         terminateInstance(initialMaster);
         assertClusterSizeEventually(2, nextMaster, slave);
 
-        assertTrueAllTheTime(new AssertTask() {
-            @Override
-            public void run() {
-                assertFalse(getPartitionService(nextMaster).isMemberStateSafe());
-                assertFalse(getPartitionService(slave).isMemberStateSafe());
-            }
+        assertTrueAllTheTime(() -> {
+            assertFalse(getPartitionService(nextMaster).isMemberStateSafe());
+            assertFalse(getPartitionService(slave).isMemberStateSafe());
         }, 5);
 
         resetPacketFiltersFrom(nextMaster);
 
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() {
-                assertTrue(getPartitionService(nextMaster).isMemberStateSafe());
-                assertTrue(getPartitionService(slave).isMemberStateSafe());
-            }
+        assertTrueEventually(() -> {
+            assertTrue(getPartitionService(nextMaster).isMemberStateSafe());
+            assertTrue(getPartitionService(slave).isMemberStateSafe());
         });
     }
 
@@ -222,20 +202,19 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
                 .setProperty(ClusterProperty.HEARTBEAT_INTERVAL_SECONDS.getName(), "1")
                 .setProperty(ClusterProperty.OPERATION_CALL_TIMEOUT_MILLIS.getName(), "3000");
 
-        final HazelcastInstance master = factory.newHazelcastInstance(config);
-        final HazelcastInstance slave1 = factory.newHazelcastInstance(config);
-        final HazelcastInstance slave2 = factory.newHazelcastInstance(config);
+        HazelcastInstance master = factory.newHazelcastInstance(config);
+        HazelcastInstance slave1 = factory.newHazelcastInstance(config);
+        HazelcastInstance slave2 = factory.newHazelcastInstance(config);
 
         assertClusterSizeEventually(3, slave1, slave2);
         warmUpPartitions(master, slave1, slave2);
 
-        fillData(master);
-        assertSizeAndDataEventually();
+        fillAndAssertData(master);
 
         // prevent migrations before adding migration listeners when slave3 joins the cluster
         changeClusterStateEventually(master, ClusterState.NO_MIGRATION);
 
-        final HazelcastInstance slave3 = factory.newHazelcastInstance(config);
+        HazelcastInstance slave3 = factory.newHazelcastInstance(config);
         assertClusterSizeEventually(4, slave1, slave2);
 
         // set migration listener to drop migration commit response after a migration to slave3 is completed
@@ -253,13 +232,10 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
         waitAllForSafeState(master, slave1, slave2, slave3);
 
         final PartitionTableView masterPartitionTable = getPartitionService(master).createPartitionTableView();
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() {
-                assertEquals(masterPartitionTable, getPartitionService(slave1).createPartitionTableView());
-                assertEquals(masterPartitionTable, getPartitionService(slave2).createPartitionTableView());
-                assertEquals(masterPartitionTable, getPartitionService(slave3).createPartitionTableView());
-            }
+        assertTrueEventually(() -> {
+            assertEquals(masterPartitionTable, getPartitionService(slave1).createPartitionTableView());
+            assertEquals(masterPartitionTable, getPartitionService(slave2).createPartitionTableView());
+            assertEquals(masterPartitionTable, getPartitionService(slave3).createPartitionTableView());
         });
 
         assertSizeAndData();
@@ -270,10 +246,10 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
         assertNoDuplicateMigrations(slave3);
     }
 
-    private void setMigrationListenerToDropCommitResponse(final HazelcastInstance master, final HazelcastInstance destination) {
+    private void setMigrationListenerToDropCommitResponse(HazelcastInstance master, HazelcastInstance destination) {
         // intercept migration complete on destination and drop commit response
         getPartitionServiceImpl(destination).setMigrationInterceptor(new MigrationInterceptor() {
-            final AtomicReference<MigrationInfo> committedMigrationInfoRef = new AtomicReference<MigrationInfo>();
+            final AtomicReference<MigrationInfo> committedMigrationInfoRef = new AtomicReference<>();
 
             @Override
             public void onMigrationStart(MigrationParticipant participant, MigrationInfo migrationInfo) {
@@ -296,20 +272,19 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
                 .setProperty(ClusterProperty.HEARTBEAT_INTERVAL_SECONDS.getName(), "1")
                 .setProperty(ClusterProperty.OPERATION_CALL_TIMEOUT_MILLIS.getName(), "3000");
 
-        final HazelcastInstance master = factory.newHazelcastInstance(config);
-        final HazelcastInstance slave1 = factory.newHazelcastInstance(config);
-        final HazelcastInstance slave2 = factory.newHazelcastInstance(config);
+        HazelcastInstance master = factory.newHazelcastInstance(config);
+        HazelcastInstance slave1 = factory.newHazelcastInstance(config);
+        HazelcastInstance slave2 = factory.newHazelcastInstance(config);
 
         assertClusterSizeEventually(3, slave1, slave2);
         warmUpPartitions(master, slave1, slave2);
 
-        fillData(master);
-        assertSizeAndDataEventually();
+        fillAndAssertData(master);
 
         // prevent migrations before adding migration listeners when slave3 joins the cluster
         changeClusterStateEventually(master, ClusterState.NO_MIGRATION);
 
-        final HazelcastInstance slave3 = factory.newHazelcastInstance(config);
+        HazelcastInstance slave3 = factory.newHazelcastInstance(config);
         assertClusterSizeEventually(4, slave1, slave2);
 
         // set migration listener to drop migration commit response after a migration to slave3 is completed
@@ -326,12 +301,9 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
         waitAllForSafeState(master, slave1, slave2);
 
         final PartitionTableView masterPartitionTable = getPartitionService(master).createPartitionTableView();
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() {
-                assertEquals(masterPartitionTable, getPartitionService(slave1).createPartitionTableView());
-                assertEquals(masterPartitionTable, getPartitionService(slave2).createPartitionTableView());
-            }
+        assertTrueEventually(() -> {
+            assertEquals(masterPartitionTable, getPartitionService(slave1).createPartitionTableView());
+            assertEquals(masterPartitionTable, getPartitionService(slave2).createPartitionTableView());
         });
 
         assertSizeAndData();
@@ -348,16 +320,15 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
                 .setProperty(ClusterProperty.HEARTBEAT_INTERVAL_SECONDS.getName(), "1")
                 .setProperty(ClusterProperty.OPERATION_CALL_TIMEOUT_MILLIS.getName(), "3000");
 
-        final HazelcastInstance master = factory.newHazelcastInstance(config);
-        final HazelcastInstance slave1 = factory.newHazelcastInstance(config);
-        final HazelcastInstance slave2 = factory.newHazelcastInstance(config);
-        final HazelcastInstance slave3 = factory.newHazelcastInstance(config);
+        HazelcastInstance master = factory.newHazelcastInstance(config);
+        HazelcastInstance slave1 = factory.newHazelcastInstance(config);
+        HazelcastInstance slave2 = factory.newHazelcastInstance(config);
+        HazelcastInstance slave3 = factory.newHazelcastInstance(config);
 
         assertClusterSizeEventually(4, slave1, slave2, slave3);
         warmUpPartitions(master, slave1, slave2, slave3);
 
-        fillData(master);
-        assertSizeAndDataEventually();
+        fillAndAssertData(master);
 
         // reject promotion commits from master to prevent promotions complete when slave3 leaves the cluster
         rejectOperationsFrom(master, F_ID, singletonList(PROMOTION_COMMIT));
@@ -377,12 +348,9 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
         waitAllForSafeState(master, slave1, slave2);
 
         final PartitionTableView masterPartitionTable = getPartitionService(master).createPartitionTableView();
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() {
-                assertEquals(masterPartitionTable, getPartitionService(slave1).createPartitionTableView());
-                assertEquals(masterPartitionTable, getPartitionService(slave2).createPartitionTableView());
-            }
+        assertTrueEventually(() -> {
+            assertEquals(masterPartitionTable, getPartitionService(slave1).createPartitionTableView());
+            assertEquals(masterPartitionTable, getPartitionService(slave2).createPartitionTableView());
         });
 
         assertSizeAndData();
@@ -399,16 +367,15 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
                 .setProperty(ClusterProperty.HEARTBEAT_INTERVAL_SECONDS.getName(), "1")
                 .setProperty(ClusterProperty.OPERATION_CALL_TIMEOUT_MILLIS.getName(), "3000");
 
-        final HazelcastInstance master = factory.newHazelcastInstance(config);
-        final HazelcastInstance slave1 = factory.newHazelcastInstance(config);
-        final HazelcastInstance slave2 = factory.newHazelcastInstance(config);
-        final HazelcastInstance slave3 = factory.newHazelcastInstance(config);
+        HazelcastInstance master = factory.newHazelcastInstance(config);
+        HazelcastInstance slave1 = factory.newHazelcastInstance(config);
+        HazelcastInstance slave2 = factory.newHazelcastInstance(config);
+        HazelcastInstance slave3 = factory.newHazelcastInstance(config);
 
         assertClusterSizeEventually(4, slave1, slave2, slave3);
         warmUpPartitions(master, slave1, slave2, slave3);
 
-        fillData(master);
-        assertSizeAndDataEventually();
+        fillAndAssertData(master);
 
         // reject promotion commits from master to prevent promotions complete when slave3 leaves the cluster
         rejectOperationsFrom(master, F_ID, singletonList(PROMOTION_COMMIT));
@@ -428,12 +395,7 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
         waitAllForSafeState(master, slave1);
 
         final PartitionTableView masterPartitionTable = getPartitionService(master).createPartitionTableView();
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() {
-                assertEquals(masterPartitionTable, getPartitionService(slave1).createPartitionTableView());
-            }
-        });
+        assertTrueEventually(() -> assertEquals(masterPartitionTable, getPartitionService(slave1).createPartitionTableView()));
 
         assertSizeAndData();
 
@@ -441,7 +403,14 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
         assertNoDuplicateMigrations(slave1);
     }
 
-    private void setMigrationListenerToPromotionResponse(final HazelcastInstance master, final HazelcastInstance destination) {
+    private void fillAndAssertData(HazelcastInstance hz) {
+        assertTrueEventually(() -> {
+            fillData(hz);
+            assertSizeAndData();
+        });
+    }
+
+    private void setMigrationListenerToPromotionResponse(HazelcastInstance master, HazelcastInstance destination) {
         getPartitionServiceImpl(destination).setMigrationInterceptor(new MigrationInterceptor() {
             @Override
             public void onPromotionComplete(MigrationParticipant participant, Collection<MigrationInfo> migrationInfos, boolean success) {
@@ -455,7 +424,7 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
     private static void assertNoDuplicateMigrations(HazelcastInstance hz) {
         TestMigrationAwareService service = getNodeEngineImpl(hz).getService(TestMigrationAwareService.SERVICE_NAME);
         List<PartitionMigrationEvent> events = service.getBeforeEvents();
-        Set<PartitionMigrationEvent> uniqueEvents = new HashSet<PartitionMigrationEvent>(events);
+        Set<PartitionMigrationEvent> uniqueEvents = new HashSet<>(events);
         assertEquals("Node: " + getAddress(hz) + ", Events: " + events, uniqueEvents.size(), events.size());
     }
 
@@ -470,12 +439,7 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
         }
     }
 
-    private static void assertPartitionStateVersionInitialized(final HazelcastInstance instance) {
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() {
-                assertThat(getPartitionService(instance).getPartitionStateVersion(), Matchers.greaterThan(0));
-            }
-        });
+    private static void assertPartitionStateVersionInitialized(HazelcastInstance instance) {
+        assertTrueEventually(() -> assertThat(getPartitionService(instance).getPartitionStateVersion(), greaterThan(0)));
     }
 }

